stackslib/src/net/api/getsortition.rs

```diff
@@ -0,0 +1,380 @@
+};
+use crate::net::httpcore::{
+    HttpRequestContentsExtensions, RPCRequestHandler, StacksHttp, StacksHttpRequest,
+    StacksHttpResponse,
+};
+use crate::net::{Error as NetError, StacksNodeState, TipRequest, MAX_HEADERS};
+use crate::util_lib::db::{DBConn, Error as DBError};
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum QuerySpecifier {
+    ConsensusHash(ConsensusHash),
+    BurnchainHeaderHash(BurnchainHeaderHash),
+    BlockHeight(u64),
+    Latest,
+}
+
+pub static RPC_SORTITION_INFO_PATH: &str = "/v3/sortition";
+
+/// Struct for sortition information returned via the GetSortition API call
+#[derive(PartialEq, Debug, Clone, Serialize, Deserialize)]
+pub struct SortitionInfo {
+    /// The burnchain header hash of the block that triggered this event.
+    #[serde(with = "prefix_hex")]
+    pub burn_block_hash: BurnchainHeaderHash,
+    /// The burn height of the block that triggered this event.
+    pub burn_block_height: u64,
+    /// This sortition ID of the block that triggered this event. This incorporates
+    ///  PoX forking information and the burn block hash to obtain an identifier that is
+    ///  unique across PoX forks and burnchain forks.
+    #[serde(with = "prefix_hex")]
+    pub sortition_id: SortitionId,
+    /// The parent of this burn block's Sortition ID
+    #[serde(with = "prefix_hex")]
+    pub parent_sortition_id: SortitionId,
+    /// The consensus hash of the block that triggered this event. This incorporates
+    ///  PoX forking information and burn op information to obtain an identifier that is
+    ///  unique across PoX forks and burnchain forks.
+    #[serde(with = "prefix_hex")]
+    pub consensus_hash: ConsensusHash,
+    /// Boolean indicating whether or not there was a succesful sortition (i.e. a winning
+    ///  block or miner was chosen).
+    pub was_sortition: bool,
+    /// If sortition occurred, and the miner's VRF key registration
+    ///  associated a nakamoto mining pubkey with their commit, this
+    ///  will contain the Hash160 of that mining key.
+    #[serde(with = "prefix_opt_hex")]
+    pub miner_pk_hash160: Option<Hash160>,
+    /// If sortition occurred, this will be the consensus hash of the burn block corresponding
+    /// to the winning block commit's parent block ptr. In 3.x, this is the consensus hash of
+    /// the tenure that this new burn block's miner will be building off of.
+    #[serde(with = "prefix_opt_hex")]
+    pub stacks_parent_ch: Option<ConsensusHash>,
+    /// If sortition occurred, this will be the consensus hash of the most recent sortition before
+    ///  this one.
+    #[serde(with = "prefix_opt_hex")]
+    pub last_sortition_ch: Option<ConsensusHash>,
+    #[serde(with = "prefix_opt_hex")]
+    /// In Stacks 2.x, this is the winning block.
+    /// In Stacks 3.x, this is the first block of the parent tenure.
+    pub committed_block_hash: Option<BlockHeaderHash>,
+}
+
+mod prefix_opt_hex {
```

Comment:
I know this is indirectly testsed by your getsortition.rs tests, but might be good to have some unit tests for just the serialize functions to make bug finding faster. Not a hard stop though by any means.

Reply:
Added in fdcaea8b91972ad721d9d9e0579ac89fb531ddc0 (stackslib/src/net/api/tests/mod.rs)